magento/magento2
# Fix for wildcard section reload being broken by section-config refacor

### Description (*)
Since the refactor of section-config.js in Magento_Customer (https://github.com/magento/magento2/commit/976302de9187d12d9ed0b54e06de3ca8a6f37b24) reloading of all customer-data sections via wildcard `*` has caused requests to `customer/section/load` to respond `400`. 

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/1484134/87251411-831b3600-c439-11ea-9279-c32ea03fae1d.png">

This causes issues when you call reload directly from `customer-data` or you request wildcard section reloading via an action in sections.xml

You can validate this yourself by executing the following code from your devtools console:

```js
require('Magento_Customer/js/customer-data').reload('*');
``` 


I've added an additional condition to the `getFromServer` method's parameters variable that checks to see if the wildcard section request is present, if it is, it makes sure parameters gets set to `[]` instead of `['*']` which fails.

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/1484134/87251568-a2ff2980-c43a-11ea-80c8-9e40ad500732.png">


### Related Pull Requests

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/28428
2. Fixes https://github.com/magento/magento2/issues/28154

### Manual testing scenarios (*)

1. In your Browser's devtools, execute the following code:
```js
require('Magento_Customer/js/customer-data').reload('*');
```
This should return all customer-data sections.
2. Hitting any route with a request where the action expects to reload '*'.

### Questions or comments


### Contribution checklist (*)
